framework/src/modules/chain/modules/rounds.js

```diff
@@ -62,6 +64,128 @@ class Rounds {
 	}
 }
 
+__private.updateSendersRoundInformationWithAmountForTransactions = function(
+	round,
+	transactions,
+	forwardTick,
+	tx
+) {
+	return transactions.map(transaction => {
+		const value = new Bignumber(transaction.amount).plus(transaction.fee);
+		const valueToUpdate = forwardTick ? `-${value}` : `${value}`;
+		return self.createRoundInformationWithAmount(
+			transaction.senderId,
+			round,
+			valueToUpdate,
+			tx
+		);
+	});
+};
+
+__private.updateRecipientsRoundInformationWithAmountForTransactions = function(
+	round,
+	transactions,
+	forwardTick,
+	tx
+) {
+	// For IN_TRANSFER TRANSACTION, we first need to get the authorId to get the recipient
+	return Promise.all(
+		transactions
+			.filter(transaction => transaction.type === transactionTypes.IN_TRANSFER)
+			.map(transaction =>
+				// Check the owner of the dapp for recipient account
+				library.storage.entities.Transaction.getOne(
+					{
+						id: transaction.asset.inTransfer.dappId,
+					},
+					{},
+					tx
+				).then(dappTransaction => ({
+					...transaction,
+					recipientId: dappTransaction.senderId,
+				}))
+			)
+	).then(inTransferTransactionsWithRecipientId =>
+		// We need to only update recipient accounts for transfer transaction, in trasfer transaction and out transfer transaction
+		[
+			...transactions.filter(
+				transaction =>
+					transaction.type === transactionTypes.SEND ||
+					transaction.type === transactionTypes.OUT_TRANSFER
+			),
+			...inTransferTransactionsWithRecipientId,
+		].map(transaction => {
+			const value = transaction.amount;
+			const valueToUpdate = forwardTick ? `${value}` : `-${value}`;
+			return self.createRoundInformationWithAmount(
+				transaction.recipientId,
+				round,
+				valueToUpdate,
+				tx
+			);
+		})
+	);
+};
+
+__private.updateRoundInformationWithDelegatesForTransactions = function(
+	round,
+	transactions,
+	forwardTick,
+	tx
+) {
+	return transactions
+		.filter(transaction => transaction.type === transactionTypes.VOTE)
+		.map(transaction =>
+			(forwardTick
+				? transaction.asset.votes
+				: Diff.reverse(transaction.asset.votes)
+			).map(vote => {
+				// Fetch first character
+				const mode = vote[0];
+				const dependentId = vote.slice(1);
```

Comment:
This was the old logic to get `dependentId`, its not safe to replace with simply `slice`.

```
let mode = updatedValueItem[0];
let dependentId = '';

if (mode === '-' || mode === '+') {
	dependentId = updatedValueItem.slice(1);
} else {
	dependentId = updatedValueItem;
	mode = '+';
}
```

Reply:
At this point, every transaction has to be valid in all the case. Therefore, it should be safe to do this.
Also, it's kinda scary to update the `mode` if it's not + or -....